tests/kernel_bundle/get_kernel_bundle_without_kernel_attr_verify_that_exception_is_not_thrown.cpp

```diff
@@ -0,0 +1,133 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  This test calls these sycl::get_kernel_bundle overloads:
+//    1) sycl::get_kernel_bundle<BundleState>(context, devices, selector)
+//    2) sycl::get_kernel_bundle<BundleState>(context, selector)
+//    3) sycl::get_kernel_bundle<BundleState>(context, devices)
+//    4) sycl::get_kernel_bundle<BundleState>(context, kernelIds)
+//    5) sycl::get_kernel_bundle<BundleState>(context, devices, kernelIds)
+//    6) sycl::get_kernel_bundle<BundleState>(context)
+//    7) sycl::get_kernel_bundle<KernelName, BundleState>(context)
+//    8) sycl::get_kernel_bundle<KernelName, BundleState>(context, devices)
+//  For kernels that uses fp16 and fp64 datatypes, kernel that uses atomic_ref
+//  (it requires atomic64 aspect) and for two kernels without any requirements
+//  for device from chosen platform.
+//
+//  The test verifies that the exception not thrown for bundle_state::input and
+//  bundle_state::object.
+//
+*******************************************************************************/
+
+#include "get_kernel_bundle.h"
+
+#define TEST_NAME \
+  get_kernel_bundle_without_kernel_attr_verify_that_exception_is_not_thrown
+
+namespace TEST_NAMESPACE {
+using namespace sycl_cts;
+using namespace kernels;
+using namespace sycl_cts::tests::kernel_bundle;
+using namespace sycl_cts::tests::get_kernel_bundle;
+
+/** @brief Observing kernel bundle with provided bundle state for each overload
+ *         that don't receive KernelName template parameter the then call
+ *         "for_all_types" with "verify_that_kernel_in_bundle" function for
+ *         check existence of each predefined kernel in result bundle
+ *  @tparam BundleState sycl::bundle_state enum's field
+ *  @param log sycl_cts::util::logger class object
+ *  @param queue sycl::queue class object
+ */
+template <sycl::bundle_state BundleState>
+void run_test_for_all_overload_types(
+    util::logger &log, sycl::queue &queue,
+    const std::vector<sycl::kernel_id> &user_defined_kernel_ids) {
+  const auto context = queue.get_context();
+  const auto device = queue.get_device();
+
+  const auto selector = [](const sycl::device_image<BundleState> &) {
+    return true;
+  };
+
```

Comment:
I am starting to feel that some code should be factorized out among all these tests...
Like this `always_device_image_selector`.

Reply:
this comment was applied [here](https://github.com/KhronosGroup/SYCL-CTS/pull/224/commits/c151a8cdae55ff2813aa9db1ffde34b33ebd1c11#diff-cfce662345b154bd62edfd6bfceb4dc9e924282661c3234f57122c0262abe876R75-R76)